src/core/src/graph_util.cpp

```diff
@@ -43,8 +43,11 @@ void clone_ov_nodes(const std::vector<std::shared_ptr<ov::Node>>& nodes,
                 }
             }
             auto cloned_node = node->copy_with_new_inputs(cloned_args, cloned_dependencies);
-            // There is a friendly name for this node so copy it
-            cloned_node->set_friendly_name(node->get_friendly_name());
+            // generally, if there is a friendly name for this node, copy it
+            // but if the node is auto-generated, do not need to carry on the friendly name
+            // this is important to get aligned hash value for cloned model
+            if (node->get_friendly_name() != node->get_name())
+                cloned_node->set_friendly_name(node->get_friendly_name());
```

Comment:
@ilya-lavrenov can you check if the friendly name is correctly manipulated here? or other situations that may break?